diff --color -u -r a/exfat_oal.c b/exfat_oal.c
--- a/exfat_oal.c	1970-01-01 08:00:01.000000000 +0800
+++ b/exfat_oal.c	2020-06-22 01:22:19.536587993 +0800
@@ -121,7 +121,7 @@
 	} while (0)
 
 /* Linear day numbers of the respective 1sts in non-leap years. */
-static time_t accum_days_in_year[] = {
+static ktime_t accum_days_in_year[] = {
 	/* Jan  Feb  Mar  Apr  May  Jun  Jul  Aug  Sep  Oct  Nov  Dec */
 	0,   0,  31,  59,  90, 120, 151, 181, 212, 243, 273, 304, 334, 0, 0, 0,
 };
@@ -133,7 +133,7 @@
 #else
 	struct timespec64 ts;
 #endif
-	time_t second, day, leap_day, month, year;
+	ktime_t second, day, leap_day, month, year;
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4,8,0)
 	ts = CURRENT_TIME_SEC;
diff --color -u -r a/exfat_super.c b/exfat_super.c
--- a/exfat_super.c	1970-01-01 08:00:01.000000000 +0800
+++ b/exfat_super.c	2020-06-22 01:22:19.537588053 +0800
@@ -144,7 +144,7 @@
 	} while (0)
 
 /* Linear day numbers of the respective 1sts in non-leap years. */
-static time_t accum_days_in_year[] = {
+static ktime_t accum_days_in_year[] = {
 	/* Jan  Feb  Mar  Apr  May  Jun  Jul  Aug  Sep  Oct  Nov  Dec */
 	0,   0,  31,  59,  90, 120, 151, 181, 212, 243, 273, 304, 334, 0, 0, 0,
 };
@@ -155,8 +155,8 @@
 void exfat_time_fat2unix(struct exfat_sb_info *sbi, struct timespec_compat *ts,
                                                  DATE_TIME_T *tp)
 {
-	time_t year = tp->Year;
-	time_t ld;
+	ktime_t year = tp->Year;
+	ktime_t ld;
 
 	MAKE_LEAP_YEAR(ld, year);
 
@@ -174,9 +174,9 @@
 void exfat_time_unix2fat(struct exfat_sb_info *sbi, struct timespec_compat *ts,
 						 DATE_TIME_T *tp)
 {
-	time_t second = ts->tv_sec;
-	time_t day, month, year;
-	time_t ld;
+	ktime_t second = ts->tv_sec;
+	ktime_t day, month, year;
+	ktime_t ld;
 
 	second -= sys_tz.tz_minuteswest * SECS_PER_MIN;
 
